buffer/pool.go

```diff
@@ -0,0 +1,27 @@
+package buffer
+
+import (
+	"sync"
+)
+
+type Pool struct {
+	buffers sync.Pool
```

Comment:
In the [docs for sync.Pool](https://golang.org/pkg/sync/#Pool) there is a note: "A Pool must not be copied after first use." I think we should also add "must not be copied" as a documentation item on this struct.

Reply:
Thanks, I have added it in a new commit.

BTW, actually the `go vet` can detect the copy. Try `go vet` on this code:
```golang
package main

import (
	"github.com/cloudflare/cloudflared/buffer"
)

func main() {
	poolPtr := buffer.NewPool(512 * 1024)
	copy := *poolPtr

	copy.Get()
}
```
it will output:
```shell
▶ go vet main.go
# command-line-arguments
./main.go:9:10: assignment copies lock value to copy: github.com/cloudflare/cloudflared/buffer.Pool contains sync.Pool contains sync.noCopy
```

Ref: https://golang.org/issues/8005#issuecomment-190753527